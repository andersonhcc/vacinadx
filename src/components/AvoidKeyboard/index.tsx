import React from 'react';

import {Container} from './styles';
import {AvoidKeyboardProps} from './types';

const AvoidKeyboard = ({children}: AvoidKeyboardProps) => {
  return <Container>{children}</Container>;
};

export {AvoidKeyboard};
