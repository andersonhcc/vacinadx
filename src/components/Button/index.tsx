import React, {useContext, useMemo} from 'react';
import {ButtonProps} from './types';
import {ThemeContext} from 'styled-components';
import {Container, Title, Loading, AbsoluteIcon} from './styles';

const Button = ({
  children,
  color = 'primary',
  typography = 'body2',
  loading,
  icon,
  mode = 'contained',
  onPress,
  ...rest
}: ButtonProps) => {
  const {colors} = useContext(ThemeContext);

  const colorByMode = useMemo(() => {
    return mode === 'outlined' ? colors[color].main : colors[color].onMain;
  }, [mode, color, colors]);

  return (
    <Container
      mode={mode}
      onPress={onPress}
      borderColor={colors[color].main}
      color={colors[color].main}
      {...rest}>
      {!!icon && <AbsoluteIcon>{icon}</AbsoluteIcon>}
      <Title typography={typography} color={colorByMode}>
        {children}
      </Title>
      {loading && <Loading size={15} color={colorByMode} />}
    </Container>
  );
};

export {Button};
