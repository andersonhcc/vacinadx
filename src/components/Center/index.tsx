import React, {ReactNode} from 'react';
import {Container} from './styles';

type Props = {
  children: ReactNode;
};

const Center: React.FC<Props> = ({children}: Props) => {
  return <Container>{children}</Container>;
};

export {Center};
