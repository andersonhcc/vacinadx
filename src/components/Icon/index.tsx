import React from 'react';
import {Container} from './styles';
import {IconProps} from './types';
import Icons from '~/constants/icon';

const Icon = ({
  icon,
  size = 20,
  activeColor,
  style,
}: Omit<IconProps, 'source'>) => {
  if (activeColor) {
    return (
      <Container
        size={size}
        source={Icons[icon]}
        style={[{tintColor: activeColor}, style]}
        resizeMode="contain"
      />
    );
  }
  return <Container size={size} source={Icons[icon]} resizeMode="contain" />;
};

export default Icon;
