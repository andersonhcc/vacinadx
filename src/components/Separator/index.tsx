import React from 'react';
import {Props} from './types';

import {Container} from './styles';

const Separator = ({height, width}: Props) => {
  return <Container testID="separator" height={height} width={width} />;
};
export {Separator};
