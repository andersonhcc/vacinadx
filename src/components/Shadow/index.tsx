import React from 'react';
import {Pressable} from 'react-native';
import styles from './styles';
import {ShadowProps} from './types';

const Shadow = ({children, ...rest}: ShadowProps) => {
  return (
    <Pressable style={styles.shadow} {...rest}>
      {children}
    </Pressable>
  );
};

export {Shadow};
