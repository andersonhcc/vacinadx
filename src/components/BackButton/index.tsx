import React from 'react';
import Icon from '../Icon';

import {Container} from './styles';
import {Props} from './types';

const BackButton = ({icon, ...rest}: Props) => {
  return (
    <Container {...rest}>
      <Icon icon={icon} size={15} />
    </Container>
  );
};

export {BackButton};
