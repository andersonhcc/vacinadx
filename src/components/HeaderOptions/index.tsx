import React, {useMemo} from 'react';
import {Separator} from '../Separator';
import {StatusBar} from 'react-native';
import {useTheme} from 'styled-components';
// import {View} from 'react-native';

import {Container} from './styles';

import {Props} from './types';

const HeaderOptions = ({
  left = <Separator />,
  center = <Separator />,
  right = <Separator />,
}: Props) => {
  const {spacing} = useTheme();

  const heightSeperatorStatusBar = useMemo(() => {
    return StatusBar.currentHeight
      ? StatusBar.currentHeight + spacing.md
      : spacing.md;
  }, [spacing]);

  return (
    <>
      <Separator height={heightSeperatorStatusBar} />
      <Container>
        {left}
        {center}
        {right}
      </Container>
    </>
  );
};

export {HeaderOptions};
