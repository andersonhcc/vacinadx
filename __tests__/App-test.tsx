/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Separator} from '~/components/Separator';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const {toJSON} = renderer.create(<Separator height={20} />);

  expect(toJSON()).toMatchSnapshot();
});
