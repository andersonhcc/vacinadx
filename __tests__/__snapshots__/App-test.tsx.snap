// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  height={20}
  style={
    [
      {
        "height": 20,
        "maxHeight": 20,
        "minHeight": 20,
      },
    ]
  }
/>
`;
